test/eventinge2erekt/servicemesh/multitenant_test.go

```diff
@@ -0,0 +1,63 @@
+package servicemesh
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/cloudevents/sdk-go/v2/test"
+	eventingfeatures "github.com/openshift-knative/serverless-operator/test/eventinge2erekt/features"
+	"knative.dev/eventing/test/rekt/resources/pingsource"
+	"knative.dev/reconciler-test/pkg/environment"
+	"knative.dev/reconciler-test/pkg/eventshub"
+	"knative.dev/reconciler-test/pkg/eventshub/assert"
+	"knative.dev/reconciler-test/pkg/feature"
+	"knative.dev/reconciler-test/pkg/resources/service"
+)
+
+// PingSource (tenant-2) -> Ksvc (tenant-1) -> Sink (tenant-1)
+func TestPingSourceToKsvcCrossTenant(t *testing.T) {
```

Comment:
Might be wrong, but I am wondering if there is overlapping in tests and if we could restructure test cases to have the same common traffic graph but test it under different configurations with/without istio, with/without encryption, with/without grpc etc like in https://github.com/istio/istio/blob/master/tests/integration/pilot/common/traffic.go#L256. 
I was thinking more of covering the same path under different setups.
Btw in the future with upcoming Istio migration we will have more variation in tests as well.

Reply:
I don't think there's a big overlap.
Most of the E2E tests can be run w/ and w/o Istio, w/ and w/o internal encryption. That's what we already do, by default we specify `spec.config.network.internal-encryption=true` which means all tests are run with internal encryption. When we install full Mesh we then remove this element and run all tests again with Service Mesh.
The only exception are tests under test/eventinge2erekt/servicemesh/ and test/extensione2erekt/servicemesh  which were added in this pull request. These tests verify that traffic is blocked by Service Mesh. So they are very specific for SM and can't be used for internal encryption or GRPC.
There are also tests under test/servinge2e/servicemesh which test multitenancy. It's a mixture of positive (traffic allowed) and negative (traffic forbidden) test cases. They're again quite specific for Service Mesh and multitenancy.